magpy/gui/magpy_gui.py

```diff
@@ -3223,7 +3255,75 @@ def onFlagDropButton(self,event):
 
         self.changeStatusbar("Ready")
 
+    def onFlagMinButton(self,event):
+        """
+        DESCRIPTION
+            Flags minimum value in zoomed region
+        """
+        keys = self.shownkeylist
+        teststream = self.plotstream.copy()
+        # limits
+        self.xlimits = self.plot_p.xlimits
+        if not self.xlimits == [self.plotstream.ndarray[0],self.plotstream.ndarray[-1]]:
+            testarray = self.plotstream._select_timerange(starttime=self.xlimits[0],endtime=self.xlimits[1])
+            teststream = DataStream([LineStruct()],self.plotstream.header,testarray)
+        xdata = self.plot_p.t
+        xtol = ((max(xdata) - min(xdata))/float(len(xdata)))/2
+        mini = [teststream._get_min(key,returntime=True) for key in keys]
+        flaglist = []
+        comment = 'Flagged minimum'
+        flagid = self.menu_p.str_page.FlagIDComboBox.GetValue()
+        flagid = int(flagid[0])
+        if flagid is 0:
+            comment = ''
+        for idx,me in enumerate(mini):
+            if eval('self.menu_p.str_page.'+keys[idx]+'CheckBox.IsChecked()'):
+                starttime = num2date(me[1] - xtol)
+                endtime = num2date(me[1] + xtol)
+                flaglist.extend(self.plotstream.flag_range(keys=self.shownkeylist,flagnum=flagid,text=comment,keystoflag=keys[idx],starttime=starttime,endtime=endtime))
+        if len(flaglist) > 0:
+            self.menu_p.rep_page.logMsg('- flagged minimum: added {} flags'.format(len(flaglist)))
+            self.flaglist.extend(flaglist)
+            self.plotstream = self.plotstream.flag(flaglist)
+            self.ActivateControls(self.plotstream)
+            self.plotopt['annotate'] = True
+            self.menu_p.str_page.annotateCheckBox.SetValue(True)
+            self.OnPlot(self.plotstream,self.shownkeylist)
 
+    def onFlagMaxButton(self,event):
+        """
+        DESCRIPTION
+            Flags maximum value in zoomed region
+        """
+        keys = self.shownkeylist
+        teststream = self.plotstream.copy()
+        # limits
+        self.xlimits = self.plot_p.xlimits
+        if not self.xlimits == [self.plotstream.ndarray[0],self.plotstream.ndarray[-1]]:
+            testarray = self.plotstream._select_timerange(starttime=self.xlimits[0],endtime=self.xlimits[1])
+            teststream = DataStream([LineStruct()],self.plotstream.header,testarray)
+        xdata = self.plot_p.t
+        xtol = ((max(xdata) - min(xdata))/float(len(xdata)))/2
+        maxi = [teststream._get_max(key,returntime=True) for key in keys]
+        flaglist = []
+        comment = 'Flagged maximum'
+        flagid = self.menu_p.str_page.FlagIDComboBox.GetValue()
+        flagid = int(flagid[0])
+        if flagid is 0:
+            comment = ''
+        for idx,me in enumerate(maxi):
+            if eval('self.menu_p.str_page.'+keys[idx]+'CheckBox.IsChecked()'):
```

Comment:
Also remove this eval.

Getting an exception after I've added DeltaF to the plot, where it cannot find a dfCheckbox:
```
ValueError: attempt to get argmin of an empty sequence
Traceback (most recent call last):
  File "/Users/jmfee/Documents/git/magpy/magpy/gui/magpy_gui.py", line 3381, in onFlagMaxButton
    if eval('self.menu_p.str_page.'+keys[idx]+'CheckBox.IsChecked()'):
  File "<string>", line 1, in <module>
AttributeError: 'StreamPage' object has no attribute 'dfCheckBox'
Traceback (most recent call last):
  File "/Users/jmfee/Documents/git/magpy/magpy/gui/magpy_gui.py", line 3381, in onFlagMaxButton
    if eval('self.menu_p.str_page.'+keys[idx]+'CheckBox.IsChecked()'):
  File "<string>", line 1, in <module>
AttributeError: 'StreamPage' object has no attribute 'dfCheckBox'
```